exercises/alphametics/src/Alphametics.hs

```diff
@@ -0,0 +1,4 @@
+module Alphametics (solve) where
+
+solve :: String -> Maybe [(Char, Int)]
+solve puzzle = undefined
```

Comment:
I thought so, too. One can implement a solution using a `Map`, say, and then easily transform to this basic type.
About including `puzzle` I considered
```haskell
type Puzzle = String

solve :: Puzzle -> Maybe [(Char, Int)]
solve = undefined
```
but decided against it as this coding style with `type`s might not be to everybody's liking.